recipes/cppcheck/all/conanfile.py

```diff
@@ -0,0 +1,66 @@
+"""Conan recipe package for cppcheck
+"""
+import os
+from conans import ConanFile, CMake, tools
+
+
+class CppcheckConan(ConanFile):
+    name = "cppcheck"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/danmar/cppcheck"
+    topics = ("Cpp Check", "static analyzer")
+    description = "Cppcheck is an analysis tool for C/C++ code."
+    license = "GPL-3.0-or-later"
+    generators = {"cmake"}
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"with_z3": [True, False], "have_rules": [True, False]}
+    default_options = {"with_z3": True, "have_rules": True}
+    exports_sources = ["CMakeLists.txt", "patches/**"]
+    
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+        
+    def _patch_sources(self):
+        for patch in self.conan_data["patches"][self.version]:
+            tools.patch(**patch)
+
+    def requirements(self):
+        if self.options.with_z3:
+            self.requires("z3/4.8.8")
+        if self.options.have_rules:
+            self.requires("pcre/8.44")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = "cppcheck-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        cmake.definitions["USE_Z3"] = self.options.with_z3
```

Comment:
https://github.com/danmar/cppcheck/blob/488813d00fab8278a78d518cf0d3d6651c8f0caa/cmake/options.cmake#L38
```suggestion
        cmake.definitions["ENABLE_OSS_FUZZ"] = False
        cmake.definitions["USE_Z3"] = self.options.with_z3
```